reference/test/preconditioner/batch_ilu_kernels.cpp

```diff
@@ -223,25 +373,34 @@ TYPED_TEST(BatchIlu, ExactTrsvAppliesToSingleVector)
     auto umtxs = gko::test::share(this->mtx->unbatch());
     auto ub = b->unbatch();
     auto ux = x->unbatch();
-    auto unbatch_prec = unbatch_prec_type::build().on(exec);
-    auto unbatch_fact = unbatch_fact_type::build().on(exec);
-    std::vector<mat_data_type> l_factors_data(umtxs.size());
-    std::vector<mat_data_type> u_factors_data(umtxs.size());
+
+    auto unbatch_factorization_fact =
+        gko::share(unbatch_fact_type::build()
+                       .with_iterations(static_cast<gko::size_type>(sweeps))
+                       .with_skip_sorting(true)
```

Comment:
BatchCsr does not have a sorting kernel yet, so adding a test for non-sorting input matrix would result in test failure.

Reply:
That is true. I guess we should note that here as a TODO maybe so that we remember to fix it later. It should be simple enough. To just sort the col_idxs and one set of values and use the exchange indices to sort the rest of the value vector in parallel at the same time.

Reply:
Marked as a TODO